tests/components/gree/test_switch.py

```diff
@@ -0,0 +1,155 @@
+"""Tests for gree component."""
+from datetime import timedelta
+
+from greeclimate.exceptions import DeviceTimeoutError
+import pytest
+
+from homeassistant.components.gree.const import DOMAIN as GREE_DOMAIN
+from homeassistant.components.switch import DOMAIN
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_FRIENDLY_NAME,
+    SERVICE_TOGGLE,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+ENTITY_ID = f"{DOMAIN}.fake_device_1_panel_light"
+
+
+@pytest.fixture
+def mock_now():
+    """Fixture for dtutil.now."""
+    return dt_util.utcnow()
+
+
+async def async_setup_gree(hass):
+    """Set up the gree switch platform."""
+    MockConfigEntry(domain=GREE_DOMAIN).add_to_hass(hass)
+    await async_setup_component(hass, GREE_DOMAIN, {GREE_DOMAIN: {DOMAIN: {}}})
+    await hass.async_block_till_done()
+
+
+async def test_send_panel_light_on(hass, discovery, device, mock_now):
+    """Test for sending power on command to the device."""
+    await async_setup_gree(hass)
+
+    next_update = mock_now + timedelta(minutes=5)
```

Comment:
Optional: My read of this test file is that every test needs this 5 minute alarm to fire in setup.  Consider extracting to a helper function.

Aside: While I am reading this, I am wondering, why is this 5 minute timer needed for setup of these tests? If you do end up extracting to a helper function, it could also explain the rationale for this as a standard test setup.  (FWIW I started this review by reading the tests, so i expect to learn why this is needed once I read the code)

Reply:
Honestly not even sure if this is necessary ... I'll check.

Reply:
Not needed, removed the unnecessary lines from all the test cases.